src/gui/editorwidgets/qgsvaluerelationwidgetwrapper.cpp

```diff
@@ -69,7 +69,22 @@ QVariant QgsValueRelationWidgetWrapper::value() const
         }
       }
     }
-    v = selection.join( QStringLiteral( "," ) ).prepend( '{' ).append( '}' );
+
+    if ( layer()->fields().at( fieldIdx() ).type() == QVariant::Map )
+    {
+      QVariantList vl;
+      //store as QVariantList because it's json
+      for ( const QString &s, qgis::as_const( selection ) )
+      {
+        vl << s;
+      }
+      v = vl;
+    }
+    else
+    {
+      //store as hstore string
+      v = selection.join( QStringLiteral( ',' ) ).prepend( '{' ).append( '}' );
```

Comment:
More a question than a comment: is `QStringLiteral` really useful here?

Reply:
Not actually, it's a leftover of trying to fix things online https://github.com/qgis/QGIS/pull/8721/commits/eb9310fec88c851e8a147e8b6ef0480756e51342 ;P